### ISSUE TITLE
Encountered an Error while Syncing: Error: Account is missing required fields:imap_*

### ISSUE THREAD

**Issue by rawrjustin** [Foundry376/Mailspring#56]
##### Are there any related issues?
Can't find any related issues.

##### What operating system are you using?
Mac

##### What version of Mailspring are you using?
1.0.1

--

**Bug?**
##### Do you have any third-party plugins installed? If so, which ones?
No plugins
##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
Gmail

Logs:
```
Error: SyncWorker crashed with null (code 1):
Error: Account is missing required fields:imap_* Objectlog: "74806 [2017-10-06 10:50:26.520] [foreground] [info] 🔸 Updating message 39453=39453 with subject: The secret lives of children and their phones↵74806 [2017-10-06 10:50:26.521] [foreground] [info] 🔸 Updating message 39454=39454 with subject: Re: List of Error States↵74806 [2017-10-06 10:50:26.522] [foreground] [info] 🔸 Updating message 39455=39455 with subject: [Product/Beta Alert] Delayed chat messages↵74806 [2017-10-06 10:50:26.523] [foreground] [info] 🔸 Updating message 39456=39456 with subject: Re: List of Error States↵74806 [2017-10-06 10:50:26.524] [foreground] [info] 🔸 Updating message 39457=39457 with subject: Re: The secret lives of children and their phones↵74806 [2017-10-06 10:50:26.525] [foreground] [info] 🔸 Updating message 39458=39458 with subject: Jerry Li sent you a 1-1 message↵74806 [2017-10-06 10:50:26.526] [foreground] [info] 🔸 Updating message 39459=39459 with subject: Canteen Lunch Menu!↵74806 [2017-10-06 10:50:26.539] [foreground] [info] 🔸 Updating message 39460=39460 with subject: [jira]- (CREATE-6970) rawrjustinn - No keyboard when editing snap↵"provider: "gmail"stack: undefined__proto__: Objectconstructor: function Object()hasOwnProperty: function hasOwnProperty()isPrototypeOf: function isPrototypeOf()propertyIsEnumerable: function propertyIsEnumerable()toLocaleString: function toLocaleString()toString: function toString()valueOf: function valueOf()__defineGetter__: function __defineGetter__()__defineSetter__: function __defineSetter__()__lookupGetter__: function __lookupGetter__()__lookupSetter__: function __lookupSetter__()get __proto__: function __proto__()set __proto__: function __proto__()
module.exports.ErrorLogger.reportError @ /Applications/Mailspring.app/Contents/Resources/app.asar/src/error-logger.js:114
internal/process/warning.js:21 (node:75149) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 51 trigger listeners added. Use emitter.setMaxListeners() to increase limit
process.on @ internal/process/warning.js:21
```

**Comment by rawrjustin** [comment]
The account will work for a few seconds everytime I press "Try Again" in the bottom left, then it wont work at all again and makes the client unusable for any actions (receiving emails/sending/archiving/etc)

**Comment by emadow** [comment]
Also seeing this error

**Comment by Greyvend** [comment]
Same issue occurs on Ubuntu 16.04, Mailspring 1.0.4, with Gmail.

**Comment by codedmart** [comment]
Seeing this issue on ArchLinux, Mailspring 1.0.4, with IMAP and Gmail.

**Comment by bengotow** [comment]
Hey folks—there are a bunch of reasons you can see errors while syncing, and most of the time the sync process just silently restarts, reports the error back to our servers, and you can carry on with whatever you were doing. 

This one (`Error: Account is missing required fields:imap_*`) is a bit interesting. I'm almost positive it's complaining about the `imap_password` field missing, but we'll need to improve the error message in the next version to make sure that's what's going on. I thought we'd solved the password issues on linux by requiring `libsecret` and I haven't seen them on Mac before.

Is there any chance you have a large number of accounts linked? It's possible this could happen because of secret max-lengths if you have ~10 accounts linked, but we'd need to check.

**Comment by emadow** [comment]
@bengotow -- I've got two accounts linked.

**Comment by rawrjustin** [comment]
@bengotow I've got 3 accounts linked

**Comment by codedmart** [comment]
@bengotow I have 2 accounts linked.

**Comment by bengotow** [comment]
(Closing this since it hasn't been happening since October and the relevant error message has been split into several to make debugging this easier.)

**Comment by zauan** [comment]
I know this is closed, however, it is happening in Ubuntu 20.10 and pop OS 20.10

```29944 [2020-11-16 21:50:42.836] [background] [critical] 
***
*** Mailspring Sync 
*** A segmentation fault (SIGSEGV) occurred during program execution: 
*** dereference a pointer
*** that is null or invalid.
***
